### DISCUSSION TITLE
calcGrains returning NaNs for meanOrientation from EBSD data

### DISCUSSION THREAD

**Discussion by lkuna24** [mtex-toolbox/mtex#1482] [Ask Anything] [answered]
Hi All,
I am hoping someone could help me troubleshoot why when I try to use calcGrains I am getting NaN for some of the angles in meanOrientation.  The script I am using is very simple (maybe too simple?):

```
%% Specify Crystal and Specimen Symmetries

% crystal symmetry
CS = crystalSymmetry('43')

% plotting convention
setMTEXpref('xAxisDirection','east');
setMTEXpref('yAxisDirection','south');

%% Specify File Names

% path to files
pname = 'pwd';

% which files to be imported
fname = [pname '/TestSlice4.txt'];

%% Import the Data

% create an EBSD variable containing the data
ebsd = EBSD.load(fname,CS,'ColumnNames', { 'x' 'y' 'phi1' 'Phi' 'phi2'}, 'Bunge');

grains = calcGrains(ebsd);
grains = smooth(grains,2.3);
plot(ebsd,ebsd.orientations)
hold on
plot(grains.boundary,'linewidth',4)
hold off

```
I have attatched the file which I am trying to get this to work with.  Ultimately I would like to compute SchmidFactor, but without the meanOrientations I do not see how can I can proceed.  Unless of course I can compute the SchmidFactor on ebsd.orientations, but then how can I plot it against the grains?

Thanks for your help,
Lukasz

[TestSlice4.txt](https://github.com/mtex-toolbox/mtex/files/9060508/TestSlice4.txt)

**Comment by kilir** [comment, marked as answer]
Hi Lukasz,

I am getting 
```
sum(isnan(grains.meanOrientation))
ans =

     0
```
on the latest version of mtex.

![untitled](https://user-images.githubusercontent.com/11697557/177723981-f3c2b61f-036b-411c-9dd8-98031af894b9.jpg)

Cheers,
Rüdiger

**Reply by lkuna24** [reply]
Yes, unfortunately I am using mtex-5.3 and am getting
```
>> sum(isnan(grains.meanOrientation))

ans =

    36
```

Let me try with a newer version

**Reply by lkuna24** [reply]
Well, that worked.  Thank you Rüdiger
